en/docs/develop/customizations/advanced-ui-customization.md

```diff
@@ -0,0 +1,402 @@
+
+## Making advanced UI customizations to the WSO2 API-M Publisher and Developer Portal UIs
+
+Follow the instructions below to add advanced UI customizations to the Developer Portal and/or Publisher. 
+
+!!! note "Before you Begin..."  
+    Make sure the following is installed in your system.
+        - **NodeJS** (minimum 8.12.0) - This is a platform required for ReactJS development.
+        - **NPM**(minimum 5.7.0)
+
+1. Navigate to the `<API-M_HOME>/repository/deployment/server/jaggeryapps/<APP-ROOT>` directory in a terminal and run the following command. ( `<APP-ROOT>` is publisher or devportal ). This will install the local package dependencies in the publisher/devportal applications.
+
+     ```js
+     npm ci
+     ```
+
+     
+2. Build the project (enabling hot deployment) 
+
+     Run the following command to start the npm build. Note that it will continuously watch for any changes and rebuild the project. For more information see [Building the source for hot deployment](#Building the source for hot deployment) 
+
+    ```
+    npm run build:dev
+    ```
+    !!! note "Production deployment"
+        The development build is not optimized and contains a large bundle size. Make sure to use the production build when the customizations are ready for production. Use the following command to get the production-ready build.
+        
+        ```
+        npm run build:prod
+        ```
+3. Customizing the UI 
+
+     - If you want to override a specific React component or a file from the `<WEBAPP>/source/src/` directory, you need to make the changes in the following directory by only copying the desired file(s). For more information see [Overriding the UI components](#Overriding the UI components)
+         - `<WEBAPP>/override/src`
+
+### Overriding the UI components
+
+Files added in the following folder structure in the devportal/overrides folder will override the original file at devportal/source/src/app/components/Apis/Details/Documents/Documents.jsx. See the samples below for a detailed explanation.
+
+    ```sh
+    override
+    └── src
+        ├── Readme.txt
+        └── app
+            └── components
+                └── Apis
+                    └── Details
+                        ├── Documents
+                        │   └── Documentation.jsx
+                        └── Overview.jsx
+    ```
+
+### Adding new files to the override folder
+
+    ```sh
+    override
+    └── src
+        ├── Readme.txt
+        └── app
+            └── components
+                └── Apis
+                    └── Details
+                        ├── Documents
+                        │   └── Documentation.jsx
+                        └── Overview.jsx
+                        └── NewFile.jsx
+                        
+    ```
+
+You can import the **NewFile.jsx** by adding the **AppOverride** prefix to the import command followed by the full path relative to the override directory, as shown below.
```

Comment:
is this a way to add  a new file?

Reply:
Yes. This is the way to import a new file to a JSX file in overrides folder.